### PR TITLE
support using cupy wheels as a dependency

### PR DESCRIPTION
**Pull request by kmaehashi**
This PR depends on https://github.com/cupy/cupy/pull/899.

Support CuPy wheels as a dependency.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -87,15 +91,29 @@ def reduce_requirements(key):
...
+def find_any_distribution(pkgs):
+    for pkg in pkgs:
+        try:
+            return pkg_resources.get_distribution(pkg)
+        except pkg_resources.DistributionNotFound:
+            pass
+    return None
+
+
+cupy_pkg = find_any_distribution([
```
Comment (hvy) [inline setup.py]: Do you think we should add a comment here for why we simply pick the first available distribution?

Comment (kmaehashi) [reply setup.py]: Thanks, added comment.

File: setup.py [hunk 2]
```diff
@@ -87,15 +91,29 @@ def reduce_requirements(key):
...
+    'cupy',
```
[the remark this answers is not in the document]
Comment (kmaehashi) [reply setup.py]: Thanks, fixed as suggested!

**Comment by hvy** [comment]
LGTM, @okuta do you think it is okay to merge?

**Comment by okuta** [comment]
OK. But, does CuPy provides 4 packages?
```
    'cupy-cuda90',
    'cupy-cuda80',
    'cupy-cuda75',
    'cupy-cuda70',
```

**Comment by kmaehashi** [comment]
CUDA support of other wheels:
* TensorFlow: 9.0 only (was 8.0, but upgraded in recent release)
* PyTorch: 8.0/9.0/9.1
* CNTK: 8.0 only (but seems to upgrade to 9.0 in next release)
* MXNet: 7.5/8.0/9.0/9.1

**Comment by kmaehashi** [comment]
After offline discussion with @okuta san, we decided NOT to provide wheels for CUDA 7.x.

* CUDA 7.0 / 7.5 was released in 2015 and is outdated.
* Users of CUDA 7.0 / 7.5 can still install CuPy via source.

We will provide CUDA 8.0 wheel for Ubuntu 14.04 users, as CUDA 9.0 drops support for it.

**Comment by hvy** [comment]
jenkins, test this please.

**Comment by hvy** [comment]
@kmaehashi Could you check the CI failure?

**Comment by hvy** [comment]
jenkins, test this please.

**Comment by hvy** [comment]
jenkins, test this please.

**Comment by hvy** [comment]
@kmaehashi Sorry for the delay but could you resolve the conflict?

**Comment by kmaehashi** [comment]
Resolved!

**Comment by hvy** [comment]
jenkins, test this please.

**Comment by hvy** [comment]
jenkins, test this please.

**Comment by hvy** [comment]
jenkins, test this please.

**Comment by hvy** [comment]
LGTM!

**Pull request merged**